prusti-common/src/vir/optimizations/methods/cfg_cleaner.rs

```diff
@@ -0,0 +1,50 @@
+use std::collections::HashMap;
+
+use vir::{self, cfg};
+
+/// Merge consequitive basic blocks that are joined by only one edge.
+pub fn clean_cfg(mut method: cfg::CfgMethod) -> cfg::CfgMethod {
+    let predecessors = method.predecessors();
+    let traversal_order = method.get_topological_sort();
+    assert_eq!(traversal_order[0].block_index, 0, "The start block should be first.");
+    let mut new_basic_blocks = Vec::new();
+    let mut basic_blocks: HashMap<_, _> = method.basic_blocks.into_iter().enumerate().collect();
+    let mut new_indices = HashMap::new();
+
+    for cfg::CfgBlockIndex{ block_index, .. } in traversal_order {
+        if let Some(mut basic_block) = basic_blocks.remove(&block_index) {
+            let new_index = new_basic_blocks.len();
+            assert!(new_indices.insert(block_index, new_index).is_none());
+            while let vir::cfg::Successor::Goto(target) = &basic_block.successor {
+                // If the current basic block has only one successor.
+                if predecessors[&target.block_index].len() == 1 {
+                    // If the successor block has only one predecessor.
```

Comment:
These comment seem a bit confusing (to me); it looks like the "If the successor block has only one predecessor." one is about the assertion. Move each one line up?